src/index.js

```diff
@@ -249,17 +249,26 @@ class ReactNativeModal extends Component {
           animEvt = this.createAnimationEventForSwipe();
         }
 
-        if (this.isSwipeDirectionAllowed(gestureState)) {
+        // needed for when a user switches directions without ending their touch
+        this.currentSwipingDirection = this.getSwipingDirection(gestureState);
```

Comment:
This line introduces subtle bugs i've seen while testing and checking. When you swipe the modal and change direction you might sometimes be able to move the modal in unwanted directions. I didn't investigate a lot into the why but we should focus on one problem at a time. I suggest we let the swiping direction as initialized and disallow from changing direction in between, and later fix that problem to allow direction switching in the middle of a gesture.